src/engine/bank02.asm

```diff
@@ -1,17 +1,1775 @@
-Func_8000: ; 8000 (2:4000)
-	INCROM $8000, $8211
+_DuelCheckInterface: ; 8000 (2:4000)
+	call ResetCursorPosAndBlink
+	xor a
+	ld [wce5e], a
+	call DrawWideTextBox
+
+; reset cursor blink
+	xor a
+	ld [wDuelCursorBlinkCounter], a
+	ld hl, CheckMenuData
+	call PlaceTextItems
+.loop
+	call DoFrame
+	call HandleDuelMenuInput_YourPlayArea
+	jr nc, .loop
+	cp $ff
+	ret z ; B pressed
+
+; A was pressed
+	ld a, [wCursorDuelYPosition] 
+	sla a
+	ld b, a
+	ld a, [wCursorDuelXPosition]
+	add b
+	ld hl, .table
+	call JumpToFunctionInTable
+	jr _DuelCheckInterface
+
+.table: ; 8031 (2:4031)
+	dw DuelCheckMenu_InPlayArea
+	dw DuelCheckMenu_Glossary
+	dw DuelCheckMenu_YourPlayArea
+	dw DuelCheckMenu_OppPlayArea
+
+DuelCheckMenu_InPlayArea: ; 8039 (2:4039)
+	xor a
+	ld [wce60], a
+	farcall Func_180d5
+	ret
+
+DuelCheckMenu_Glossary: ; 8042 (2:4042)
+	farcall Func_006_44c8
+	ret
+
+DuelCheckMenu_YourPlayArea: ; 8047 (2:4047)
+	call ResetCursorPosAndBlink
+	xor a
+	ld [wce5e], a
+	ldh a, [hWhoseTurn]
+.draw
+	ld h, a
+	ld l, a
+	call DrawYourOrOppPlayArea_LoadTurnHolders
+
+	ld a, [wCursorDuelYPosition]
+	sla a
+	ld b, a
+	ld a, [wCursorDuelXPosition]
+	add b
+	ld [wLastCursorPosition_YourPlayArea], a
+	ld b, $f8 ; black arrow tile
+	call DrawByteToTabulatedPositions
+
+	call DrawWideTextBox
+	
+; reset cursor blink
+	xor a
+	ld [wDuelCursorBlinkCounter], a
+	ld hl, YourPlayAreaMenuData
+	call PlaceTextItems
+
+.loop
+	call DoFrame
+	xor a
+	call DrawArrowsToTabulatedPositions
+	call HandleDuelMenuInput_PlayArea
+	jr nc, .loop
+
+	call EraseByteFromTabulatedPositions
+	cp $ff
+	ret z
+
+	ld a, [wCursorDuelYPosition]
+	sla a
+	ld b, a
+	ld a, [wCursorDuelXPosition]
+	add b
+	ld hl, .table
+	call JumpToFunctionInTable
+	jr .draw
+
+.table ; 8098 (2:4098)
+	dw OpenDuelScreen.turn_holder_play_area
+	dw OpenDuelScreen.turn_holder_hand
+	dw OpenDuelScreen.turn_holder_discard_pile
+	
+OpenDuelScreen: ; 809e (2:409e)
```

Comment:
While I think the ``OpenDuelScreen`` name is spot at first glance on due to being generic duel screens triggered from the Check menu, I would still suggest to use global labels rather than local for jumptable entries. For example, ``OpenDuelScreen_TurnHolderPlayArea``, ``OpenDuelScreen_NonTurnHolderPlayArea``, etc.